src/Libraries/Nop.Data/EntityRepository.cs

```diff
@@ -602,21 +603,27 @@ public virtual async Task DeleteAsync(IList<TEntity> entities, bool publishEvent
             if (entities == null)
                 throw new ArgumentNullException(nameof(entities));
 
-            if (entities.OfType<ISoftDeletedEntity>().Any())
```

Comment:
I think it's probably better to leave one check, since TEntity will always be of the same type and check this every time for an entity in a loop - redundant code.

Reply:
Sorry, I did not see the second commit right away, the question is no longer relevant

Reply:
Yeah, the original loop with double type checking fooled me there - type constraints are very nice in this kind of situation.